microservices/first/utils/db_manager.py

```diff
@@ -0,0 +1,133 @@
+import json
+import sqlite3
+from sqlite3 import Error
+from typing import Any, Dict, List, Optional, Union
+
+from microservices.first.utils import deployment_status
+
+"""
+Deployment Table Schema:
+id: Integer, Primary key identifying each row
+deployment_id: Integer, Indentifier of deployment
+repo_name: TEXT, Name of repo the deployment is working on
+repo_branch: TEXT, Branch of the repo being worked on
+status: TEXT, (DeploymentStatus in text form) Status of the deployment.
+
+Results Table Schema:
+id: Integer, Primary key identifying each row
+deployment_id: Integer, Foreign Key pointing to Deployment Table "deployment_id". Deployment the result is associated with.
+result: TEXT, one test result object.
+"""
+class DBManager:
+  def __init__(self, db_file: str) -> None:
+    """
+    Initialize the database manager with the path to the database file.
+    
+    param db_file: The file path of the SQLite database.
+    """
+    self.db_file = db_file
+    self.conn = self.create_connection()
+
+  def create_connection(self) -> Optional[sqlite3.Connection]:
+    """Create a database connection to the SQLite database."""
+    conn = sqlite3.connect(self.db_file)
+    return conn
+    return conn
+
+  def close_connection(self) -> None:
+    """Close the database connection."""
+    if self.conn:
+        self.conn.close()
+
+  def create_deployments_table(self) -> None:
+    """Create the deployments table in the database."""
+    c = self.conn.cursor()
+    c.execute(
+              '''CREATE TABLE IF NOT EXISTS deployments (
+              id INTEGER PRIMARY KEY, 
+              repo_name TEXT, 
+              repo_branch TEXT,
+              status TEXT DEFAULT 'STARTED')'''
+            )
+
+  def create_results_table(self) -> None:
+    """Create the results table in the database."""
+    c = self.conn.cursor()
+    c.execute(
```

Comment:
Is `self.conn.commit()` needed after this `execute()`?

Reply:
DDL statements automatically commit.